### PR TITLE
Merge jetbrains-staging into public mainline

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andrewyuq — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt [hunk 1]
```diff
@@ -492,10 +499,46 @@ class CodeWhispererService {
         psiFile: PsiFile,
         latencyContext: LatencyContext
     ): RequestContext {
-        val fileContextInfo = getFileContextInfo(editor, psiFile)
-        val caretPosition = getCaretPosition(editor)
+        // 1. file context
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:502]: nit: Do we need the bullet points comments?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererFileContextProviderTest.kt [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+            assertThat(sut.extractSupplementalFileContextForTst(psi, context)).isEmpty()
+
+            context = aFileContextInfo(CodeWhispererKotlin.INSTANCE)
+            assertThat(sut.extractSupplementalFileContextForSrc(psi, context)).isEmpty()
+            assertThat(sut.extractSupplementalFileContextForTst(psi, context)).isEmpty()
+
+            // TODO: the following assertions will fail as in test env, psiFile.virtualFile == null @psiGist.getFileData(psiFile)
+//            context = aFileContextInfo(CodeWhispererJava.INSTANCE)
+//            assertThat(sut.extractSupplementalFileContextForSrc(psi, context)).isNotEmpty
+//            assertThat(sut.extractSupplementalFileContextForTst(psi, context)).isNotEmpty
```
Comment (manodnyab) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererFileContextProviderTest.kt:69]: nit: formatting

**Review by manodnyab — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt (hunk 1 above)
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:502]: I would prefer keeping these for readability

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererFileContextProviderTest.kt (hunk 2 above)
Comment (Will-ShaoHua) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererFileContextProviderTest.kt:69]: let me reformat this, in fact I am thinkging to remove them

**Pull request merged**